src/components/GitGraph.tsx

```diff
@@ -1,12 +1,14 @@
 import { Gitgraph } from "@gitgraph/react";
 import { getCommits } from "../services/apiService";
 import { ICommit } from "../models";
-import { useMutation, useQuery } from "react-query";
-import { useQueryClient } from "@tanstack/react-query";
+import { useQuery } from "@tanstack/react-query";
 import ReadMore from "./ReadMore";
+import { useParams } from "@tanstack/react-router";
 
 // create an animation for the git graph
 function GitGraph() {
+  const { username } = useParams({ strict: false });
+
   const { data: commitData, isLoading, isError } = useQuery({
     queryKey: [`getCommits`],
     queryFn: () => getCommits('Hugovs2000', 'React-Project'),
```

Comment:
Assuming this won't be hardcoded in the future?

Reply:
As you now have a username param, so maybe using that to get the repos of the username and then get the commits of the selected username?

Reply:
Yes, so I'm still working on the component where the user will be able to pick the repo. I'll add that repo as another param and then use them here to get the repo data.